ansi-terminal/unix/System/Console/ANSI/Internal.hs

```diff
@@ -72,8 +73,16 @@ hSupportsANSI :: Handle -> IO Bool
 -- (https://github.com/hspec/hspec/commit/d932f03317e0e2bd08c85b23903fb8616ae642bd)
 hSupportsANSI h = (&&) <$> hIsWritable h <*> hSupportsANSI'
  where
-  hSupportsANSI' = (&&) <$> hIsTerminalDevice h <*> isNotDumb
-  isNotDumb = (/= Just "dumb") <$> lookupEnv "TERM"
+  hSupportsANSI' = (&& isNotDumb) <$> hIsTerminalDevice h
 
 hNowSupportsANSI :: Handle -> IO Bool
 hNowSupportsANSI = hSupportsANSI
+
+-- | This function assumes that once it is first established whether or not the
+-- TERM environment variable exits with contents dumb, that will not change.
+-- This approach is taken because the use of C function setenv() in one thread
+-- can cause other threads calling C function getenv() to crash. On Unix-like
+-- operating systems, System.Environment.lookupEnv is implemented using C
+-- function getenv(). 
+isNotDumb :: Bool
+isNotDumb = unsafePerformIO (lookupEnv "TERM") /= Just "dumb"
```

Comment:
We probably want to add `{-# NOINLINE isNotDumb #-}`, otherwise GHC is free to inline it into `hSupportsANSI` and defeat our efforts to evaluate it only once.

Reply:
Now added.